src/commands/checkModel.ts

```diff
@@ -316,6 +351,71 @@ async function checkModelExists(cfgPath: string, warn = true): Promise<boolean>
     return cfgExists;
 }
 
+async function readExtendsModules(modelPath: string): Promise<string[] | undefined> {
+    try {
+        const content = await fsPromises.readFile(modelPath, 'utf8');
+        const lines = content.split(/\r?\n/);
+        for (const line of lines) {
+            const trimmed = line.trim();
+            if (trimmed.length === 0 || trimmed.startsWith('\\*')) {
+                continue;
+            }
+            if (trimmed.startsWith('EXTENDS')) {
+                const remainder = trimmed.substring('EXTENDS'.length).trim();
+                if (remainder.length === 0) {
+                    return [];
+                }
+                const uncommented = remainder
+                    .replace(/\(\*[\s\S]*?\*\)/g, ' ') // drop inline block comments
+                    .replace(/\\\*.*$/, ''); // drop trailing line comments
+                return uncommented
+                    .split(',')
+                    .map(entry => entry.trim())
+                    .filter(entry => entry.length > 0);
+            }
+            if (trimmed.startsWith('VARIABLES') || trimmed.includes('==')) {
+                break;
+            }
+        }
+    } catch (err) {
+        console.warn(`Unable to inspect model file ${modelPath}: ${err}`);
+    }
+    return undefined;
+}
+
+async function validateModelPairing(
```

Comment:
Below is a perfectly valid model that uses `INSTANCE` instead of `EXTENDS`.

```tla
----- MODULE Counter ------

CONSTANT S
VARIABLE v

Spec == v = 0 /\ [][v' \in S]_v
=====
```

```tla
----------------------------- MODULE MCCounter ------------------------------ 
VARIABLE v

INSTANCE Counter WITH S <- {1,2,3}
=============================================================================
```

```
SPECIFICATION Spec
```

Further thought and experimentation are required regarding nested modules.